scripts/lintcommit.lua

```diff
@@ -0,0 +1,35 @@
+-- Captures shell output and returns it.
+function os.capture(cmd, raw)
+  local f = assert(io.popen(cmd, 'r'))
+  local s = assert(f:read('*a'))
+  f:close()
+  if raw then return s end
+  s = string.gsub(s, '^%s+', '')
+  s = string.gsub(s, '%s+$', '')
+  s = string.gsub(s, '[\n\r]+', ' ')
+  return s
+end
+
+current_branch = os.capture("git branch --show-current")
+ancestor_commit = os.capture("git merge-base master " .. current_branch)
+commits_on_current_pr = os.capture("git rev-list " .. ancestor_commit ..".." .. current_branch)
```

Comment:
no string concatenation needed, use list form of `vim.fn.system()`

Reply:
Is there a gotcha for the `vim.fn.system()`? The script explodes when I use

```lua
ancestor_commit = vim.fn.system({'git', 'merge-base', 'master', current_branch})
```

Reply:
And by explodes I mean

![wat](https://user-images.githubusercontent.com/33953936/133146044-e22c78f9-820d-4bf1-b96d-cac603a6d3e8.png)

Reply:
I think `system()` is returning a trailing newline. Try wrapping the call in `vim.trim()`:

```lua
current_branch = vim.trim(vim.fn.system({'git', 'branch', '--show-current'})
```

Alternatively use `systemlist()` and get the first item from the result:

```lua
current_branch = vim.fn.systemlist({'git', 'branch', '--show-current'})[1]
```

Reply:
That did it. Thanks :pray:

Reply:
@gpanders have I ever mentioned how helpful you are and how awesome you are?

Reply:
I've fixed it. Any reason to avoid string concatenation in general or is it mostly because the list form is more readable?

Reply:
> I've fixed it. Any reason to avoid string concatenation in general or is it mostly because the list form is more readable?

it avoids use of the shell, which avoids quoting issues and other shell quirks. see `:help system()`